homeassistant/components/esphome/strings.json

```
@@ -50,6 +50,10 @@
     "ble_firmware_outdated": {
       "title": "Update {name} with ESPHome {version} or later",
       "description": "To improve Bluetooth reliability and performance, we highly recommend updating {name} with ESPHome {version} or later. When updating the device to ESPHome {version}, it is recommended to use a serial cable instead of an over-the-air update to take advantage of the new partition scheme."
+    },
+    "api_password_deprecated": {
+      "title": "API Password deprecated on {name}",
+      "description": "The API password for ESPHome is deprecated and we recommend you replace the password with an API encryption key."
```

We need to tell the user what to do to resolve the issue. Just a simple additional sentence or change like this can help:

```suggestion
      "description": "The API password for ESPHome is deprecated and the use of an API encryption key is recommended instead.\n\nRemove the API password and add an encryption key to your ESPHome device to resolve this issue."
```